java-connectors/kafka-connect-common/src/main/java/io/lenses/streamreactor/common/util/JarManifest.java

```diff
@@ -53,12 +53,16 @@ public class JarManifest {
   public JarManifest(URL location) {
     Manifest manifest;
 
-    try (JarFile jarFile = new JarFile(new File(location.toURI()))) {
-      manifest = jarFile.getManifest();
+    try {
+      File file = new File(location.toURI());
+      if (file.isFile()) {
+        JarFile jarFile = new JarFile(file);
```

Comment:
resource `jarFile` is never closed.

Reply:
it is now :) thanks!

Reply:
Sorry, no, there's still a potential issue with the jarFile.close() call. If an exception occurs before jarFile is closed, the file won't be closed properly, potentially leading to resource leaks. To ensure proper resource management, you should use a try-with-resources statement

```
try (JarFile jarFile = new JarFile(file)) {
                jarAttributes = Optional.ofNullable(jarFile.getManifest())
                        .flatMap(mf -> Optional.of(mf.getMainAttributes()))
                        .map(this::extractMainAttributes)
                        .orElseThrow(() -> new ConnectorStartupException("Failed to extract main attributes from Jar file."));
            }
```

Reply:
Please, don't be sorry, since it's a totally valid argument, it still can be left open in scenario you've mentioned.

Or could be, but I patched it, now it _should_ be alright ;)